bindings/python/dlite-entity-python.i

```diff
@@ -481,15 +481,20 @@ def get_instance(id: "str", metaid: "str"=None, check_storages: "bool"=True) ->
 
     def __getitem__(self, ind):
         if isinstance(ind, int):
-            return self.get_property_by_index(ind)
+            value = self.get_property_by_index(ind)
         elif self.has_property(ind):
-            return self.get_property(ind)
+            value = self.get_property(ind)
         elif isinstance(ind, int):
             raise IndexError('instance property index out of range: %d' % ind)
         else:
             raise KeyError('no such property: %s' % ind)
+        if isinstance(value, np.ndarray) and self.is_frozen():
+            value.flags.writeable = False  # ensure immutability
+        return value
```

Comment:
In the context of someone changing this array and it inadvertently also updates the instance's value due to memory-sharing, it makes sense - however, in the context of simply retrieving a value and manipulating it externally, it doesn't seem to make sense?

Reply:
This make array properties of immutable instances read-only. The user can still still access and slice the array, but not change its values. Isn't that what we want?

Reply:
> I don't think we should return a copy, since we (deliberately) are not doing it in any other case. It is very easy for the user to make a copy of the returned value if he/she wants to freely manipulate it.

Well. When returning a string or a number (int/float), it's by definition always a copy in Python. This would just mimic this behaviour. But I see your point - letting the user do their thing.
Essentially, I just wanted a situation where whatever the user does, the values stored in the frozen DLite instance _cannot_ be changed. Including when returning memory pointers to those values to the user.

> Another reason for not making a copy is that it in most cases is just a waste of memory and CPU time. If the property is a 1000x1000x1000 array we really don't want any unnecessary duplication of memory.

Fair enough.

If this implementation ensures the optimal memory usage, while keeping the value completely immutable (through the standard Python API operations), then I'm very much happy :)